internal/olm/operator/catalog_source.go

```diff
@@ -0,0 +1,97 @@
+// Copyright 2020 The Operator-SDK Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package olm
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strings"
+
+	v1alpha1 "github.com/operator-framework/api/pkg/operators/v1alpha1"
+	log "github.com/sirupsen/logrus"
+	"k8s.io/apimachinery/pkg/runtime"
+
+	olmclient "github.com/operator-framework/operator-sdk/internal/olm/client"
+	internalregistry "github.com/operator-framework/operator-sdk/internal/olm/operator/internal"
+)
+
+// CatalogResources holds resources for handling CatalogSource creation
+type CatalogResources struct {
+	namespace       string
+	registryPodName string
+	registryPodAddr string
+
+	catalogsourceObj []runtime.Object
+}
+
+// initialize will initialize catalog resources with values from
+// a registry pod and validate the CatalogResources struct
+func initialize(rp *internalregistry.RegistryPod) (*CatalogResources, error) {
```

Comment:
I don't actually see `initialize` being called anywhere. Will this be called by the manager?

Reply:
@estroz yes, this will be called by the manager

Reply:
I'd suggest using the class instead and keep a design following more the DDD principles as it was done for RegistryPod. If it will be called in some place than should not be exported? 
Also, if we do not follow the above suggestion about teh design then, I'd suggest renaming to init since is more commonly used.

Reply:
If you are just suggesting renaming, I renamed it to Init as it will called by a manager eventually.